src/cleanup_rules/swift/rules.toml

```diff
@@ -513,6 +513,155 @@ replace_node = "ternary_block"
 replace = "@block_false"
 is_seed_rule = false
 
+# delete variables declared in a function scope
+[[rules]]
+name = "delete_variable_declaration"
+query = """(
+    (function_declaration
```

Comment:
why do u need to match a `function_declaration`?

Reply:
> why do u need to match a `function_declaration`?

This is because this rule is meant to detect just the function variable declaration and not any kind of declaration.

Reply:
But then either add a constraint

```
[[rules.constraints]]
matcher = "(function_declaration) @fd"
```

Reply:
How about I do this in a follow up PR? If this works, it can save from writing different rules for adhoc declarations.